examples/gif_paintable/gif_paintable/mod.rs

```diff
@@ -0,0 +1,90 @@
+mod frame;
+mod imp;
+
+use std::io::Cursor;
+
+use frame::Frame;
+use gtk::prelude::*;
+use gtk::subclass::prelude::*;
+use gtk::{gdk, glib};
+use image::{codecs::gif::GifDecoder, AnimationDecoder};
+
+glib::wrapper! {
+    pub struct GifPaintable(ObjectSubclass<imp::GifPaintable>) @implements gdk::Paintable;
+}
+
+impl GifPaintable {
+    pub fn new() -> Self {
+        glib::Object::new(&[]).expect("Failed to create object of type GifPaintable")
+    }
+
+    // Loads the bytes of a GIF into the paintable
+    // The loading consists of decoding the gif with a GIFDecoder
+    // Then storing the frames so that the paintable can render them
+    //
+    // You can only call this function once due to the callback system in setup_next_frame
+    // causing the frames to refresh at faster intervals
+    pub fn load_from_bytes(&self, bytes: &[u8]) -> Result<(), Box<dyn std::error::Error>> {
+        let imp = self.imp();
+        imp.current_idx.set(0);
+
+        let read = Cursor::new(bytes);
+        let decoder = GifDecoder::new(read)?;
+        let frames = decoder
+            .into_frames()
+            .collect_frames()?
+            .into_iter()
+            .map(Frame::new)
+            .collect::<Vec<Frame>>();
+
+        imp.frames.replace(Some(frames));
```

Comment:
Could also store the iterator directly so that you decode frames one by one as needed?

Reply:
If we rely only on the iterator, it won't be possible to loop the animation once all the frames have been loaded because we will have consumed the iterator.

Should we load each frame as needed the first time and then use the loaded frames for the subsequent loops?

Reply:
Seems overly complicated, let's just keep it like this but maybe add a comment. In a real application you probably don't want to do this because you could easily end up loading a small GIF that decodes to multiple GB of frames :)

Reply:
We could recommend to use `GifDecoder::with_limits()`, that allows to limit the memory allocated by the GIF, or event use it in the example with a sane value.

Reply:
Sounds good to me